frontend_app_file/src/container/PopupCreateFile.jsx

```diff
@@ -122,17 +120,18 @@ class PopupCreateFile extends React.Component {
       })
       return
     }
-
-    newFileList.forEach(f => { f.percent = 0 })
-    const uploadFiles = this.state.uploadFiles
-    uploadFiles.push(...newFileList)
-    this.setState({ uploadFiles })
+    newFileList = newFileList.map(f => {
```

Comment:
use spread operator and implicit return to do the map in a single line.

next level is to use spread operator to avoid using the .push() function because it is a mutable one.

Reply:
Since newFileList is an array of [File](https://developer.mozilla.org/fr/docs/Web/API/File), we can't use spread operator on it in order to add a new attribute

Reply:
ok, then add a comment explaining it with the link please